src/server/pfs/server/driver.go

```diff
@@ -1419,62 +1436,311 @@ func (d *driver) flushRepo(ctx context.Context, repo *pfs.Repo) ([]*pfs.RepoInfo
 	}
 }
 
-func (d *driver) deleteCommit(ctx context.Context, commit *pfs.Commit) error {
-	if err := d.checkIsAuthorized(ctx, commit.Repo, auth.Scope_WRITER); err != nil {
-		return err
-	}
-	commitInfo, err := d.inspectCommit(ctx, commit, false)
-	if err != nil {
+func (d *driver) deleteCommit(ctx context.Context, userCommit *pfs.Commit) error {
+	if err := d.checkIsAuthorized(ctx, userCommit.Repo, auth.Scope_WRITER); err != nil {
 		return err
 	}
+	// Main txn: Delete all downstream commits, and update subvenance of upstream commits
```

Comment:
Is it possible for for this to get called on a commit with upstream commits? I was under the impression you had to delete from the root of a provenance chain.

Reply:
Not directly, but if you delete an input commit, and then that causes a downstream output commit with multiple inputs to get deleted, the other input commit needs to be fixed